Shieldwallgen Revamp

Esoudiere opened the pull request:
- Adds in a NanoUI.
- Fixed typo in file name.
- Commented out unusued variables / replaced redundant variables.
- Significantly weaker to explosions and melee attacks.
- Added /online variant for easy mapping.
- General improvements.
- Reduced shield power usage.

Updated it on one of my branches of Baystation12. Thought the main branch could use the revamp too. It's still rather old code and really the entire thing needs to be redone, but this is a step in the right direction (hopefully)

Esoudiere committed:
-Revamped shieldwallgen

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -9,26 +9,87 @@
+		else
+			icon_state = "Shield_Gen +a"
+
+/obj/machinery/shieldwallgen/Topic(href, href_list)
```
```
if(..())
    return 1
```

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -9,26 +9,87 @@
+				"You turn on the shield generator.", \
+				"You hear heavy droning.")
+	src.add_fingerprint(usr)
+	nanomanager.update_uis(src)
```
Fingerprints are added by ..().
Return 1 in handled hrefs to  automatically update the UI interface.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -9,26 +9,87 @@
+//		var/steps = 0
+//		var/last_check = 0
+//		var/check_delay = 10
+//		var/recalc = 0
```
Don't keep removed code around.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -9,26 +9,87 @@
+		storedpower -= rand(storedpower/2, storedpower)
+	if(2)
+		storedpower -= rand(storedpower/2, storedpower)
+	..()
```
This looks like a broken (non-existing) switch.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -9,26 +9,84 @@
+		else
+			icon_state = "Shield_Gen +a"
+	if(..())
+		return 1
```
Can't say I've ever seen a version of update_icon() return anything before.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -9,26 +9,84 @@
+			usr.visible_message("\The [usr] turned the shield generator on.", \
+				"You turn on the shield generator.", \
+				"You hear heavy droning.")
+	if(..())
```
This should be done at the top of Topic(). It's this call that checks that the user is close enough, conscious, etc.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -184,18 +224,20 @@
 	var/obj/machinery/shieldwall/F
 	var/obj/machinery/shieldwallgen/G
 	var/turf/T = src.loc
 	var/turf/T2 = src.loc
```
T = get_turf(src)
T2 = T

Esoudiere replied:
Not trying to do a full optimization, just NanoUI and redundancy.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -109,13 +147,13 @@
 			for(var/dir in list(1,2,4,8)) src.cleanup(dir)
 
 /obj/machinery/shieldwallgen/proc/setup_field(var/NSEW = 0)
 	var/turf/T = src.loc
```
`var/turf/T = get_turf(src)` or add a check to abort if src.loc isn't a turf.

PsiOmegaDelta commented on code/modules/shieldgen/shieldwallgen.dm:
```diff
@@ -294,17 +336,27 @@
+				if(prob(50))
+					G = gen_primary
+				else
+					G = gen_secondary
```
Why this `G = something` copy-paste?
The pre-change variant or `var/obj/machinery/shieldwallgen/G = prob(50) ? gen_primary : gen_secondary` on line 338 would be sufficient.

PsiOmegaDelta commented:
Not noting any other issues, save maybe the \red macros. Just needs a squash and it's all good.

Esoudiere commented:
Er, I still haven't figured out how this rebasing & squashing works tbh. It seems to have made two new commits with what was squashed, I take it that's not what is suppose to happen?

Raptor1628 commented:
Not sure they should be weak to explosions, because meteor defense is one of their few jobs.

comma commented:
@Inforsaken nope that's not how squashing works. Can try doing following if you can't figure out rebase:
1.Reset to commit before all your changes SOFT.
2.Commit all changes.
3.Push

Esoudiere committed:
Shield Revamp
-Adds in a NanoUI.
-Fixed typo in file name.
-Commented out unusued variables / replaced redundant variables.
-Significantly weaker to explosions and melee attacks.
-Added /online variant for easy mapping.
-General improvements.
-Reduced shield power usage.

PsiOmegaDelta reviewed (approved):
:+1:
